Documentation - Typos & Consistency Fixes

joey-gm opened the pull request:
Fixes some typos and formatting inconsistencies.

kimdv reviewed (changes requested):
Hi @joey-gm and welcome to SwiftSyntax! 🥳

Thanks for going trough the documentation and typos. 
I see that you have changed some files in the generated files. 
Those files are generated automatically by the [Codegenerator](https://github.com/apple/swift-syntax/tree/main/CodeGeneration). 

Those files should be changed in the corresponding template file. 
After that you navigate inside CodeGeneration folder and run `swift run generate-swiftsyntax ../Sources`

This will ensure that the generated code at update correctly. 
If there is any question feel free to comment on this PR and I will gladly help you.

joey-gm commented on Sources/SwiftSyntax/generated/syntaxNodes/SyntaxStmtNodes.swift:
```diff
@@ -1575,8 +1575,7 @@ public struct MissingStmtSyntax: StmtSyntaxProtocol, SyntaxHashable {
   
-  /// A placeholder, i.e. `<#statement#>` that can be inserted into the source code to represent the missing pattern.
-  /// This token should always have `presence = .missing`.
+  /// A placeholder, i.e. `<#statement#>` that can be inserted into the source code to represent the missing pattern./// This token should always have `presence = .missing`.
```
Seems like CodeGeneration fails to add a proper line break for this particular comment.

I tried various line break options (e.g. extra spaces) in the codegen file below without avail.

https://github.com/apple/swift-syntax/blob/5922b69d0db36e5ebdadfe7a893f00ca576cb2e6/CodeGeneration/Sources/SyntaxSupport/CommonNodes.swift#L234

joey-gm commented:
@kimdv Thanks for the guidance and instruction!

Turns out some of the typos are in the CodeGeneration files, which I have now fixed. I have re-generated the swift codes with the suggested command and all files should now be in sync.

ahoppen commented on CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxVisitorFile.swift:
```diff
@@ -18,12 +18,12 @@ import Utils
   DeclSyntax(
     """
-    /// The enum describes how the SyntaxVistor should continue after visiting
+    /// The enum describes how the SyntaxVisitor should continue after visiting
```
Now that we spell it correctly, this can be a docc link 😉 

```suggestion
    /// The enum describes how the ``SyntaxVisitor`` should continue after visiting
```

ahoppen replied:
Since you changed this file in CodeGeneration, you need to ge-generate the files again as @kimdv described in https://github.com/apple/swift-syntax/pull/1841#pullrequestreview-1497883175

ahoppen commented on Sources/SwiftParser/SwiftParserCompatibility.swift:
```diff
@@ -26,6 +26,6 @@ public extension TokenSpec {
 
 //==========================================================================//
-// IMPORTANT: If you are tempted to add a compatiblity layer code here      //
+// IMPORTANT: If you are tempted to add a compatibility layer code here      //
```
```suggestion
// IMPORTANT: If you are tempted to add a compatibility layer code here     //
```

ahoppen commented on Sources/SwiftParser/TokenPrecedence.swift:
```diff
@@ -22,7 +22,7 @@ enum TokenPrecedence: Comparable {
   /// Keywords and operators that can occur in the middle of an expression
   case exprKeyword
-  /// A token that starts a bracketet expression which typically occurs inside
+  /// A token that starts a bracket expression which typically occurs inside
```
This is meant to be `bracketed`. Same in a couple more cases in this file.

```suggestion
  /// A token that starts a bracketed expression which typically occurs inside
```

ahoppen commented on Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift:
```diff
@@ -82,6 +82,6 @@ public extension TokenSyntax {
 
 //==========================================================================//
-// IMPORTANT: If you are tempted to add a compatiblity layer code here      //
+// IMPORTANT: If you are tempted to add a compatibility layer code here      //
```
```suggestion
// IMPORTANT: If you are tempted to add a compatibility layer code here     //
```

ahoppen reviewed (commented):
Thank you for fixing all those typos @joey-gm. I’ve got a couple of comments inline, otherwise it looks good to me.

ahoppen commented on Sources/SwiftParser/TokenPrecedence.swift:
```diff
@@ -134,7 +134,7 @@ enum TokenPrecedence: Comparable {
       self = .exprKeyword
 
-    // MARK: Weak bracketet
+    // MARK: Weak bracket
```
```suggestion
    // MARK: Weak bracketed
```

ahoppen commented on Sources/SwiftParser/TokenPrecedence.swift:
```diff
@@ -158,7 +158,7 @@ enum TokenPrecedence: Comparable {
       self = .weakBracketClose
 
-    // MARK: Strong bracketet
+    // MARK: Strong bracket
```
```suggestion
    // MARK: Strong bracketed
```

ahoppen reviewed (commented):
Thanks. Two more comments.

Could you also squash your commits? It just makes for a nicer git history.

ahoppen commented on Sources/SwiftSyntax/generated/SyntaxVisitor.swift:
```diff
@@ -12,12 +12,12 @@
 //===----------------------------------------------------------------------===//
 
-/// The enum describes how the SyntaxVistor should continue after visiting
+/// The enum describes how the SyntaxVisitor should continue after visiting
```
I don’t think you re-generated SyntaxVisitor. If you did, this line would be

```swift
/// The enum describes how the ``SyntaxVisitor`` should continue after visiting
```

joey-gm commented:
> One more squash of these two commits and we’re ready to merge.

All done. Thanks!

ahoppen reviewed (approved):
Thanks. Once CI passes, I’ll merge the PR

ahoppen commented:
@swift-ci Please test

ahoppen commented:
@swift-ci Please test

ahoppen commented:
@joey-gm Unfortunately we’ve got a merge conflict now. Could you rebase your changes on top of `main`?

ahoppen commented:
@swift-ci Please test

joey-gm commented:
Hi Alex @ahoppen,

I've rebased and recommitted the changes.

Below are the steps I’ve taken. Could you please review and make sure I’ve rebased properly? (I’m an amateur and not too familiar with git’s branch management. I skimmed through the changes in the latest commit and they look okay.)

Thanks!

(Sync my `main` branch with the Apple's `main`)
git checkout documentation  // `documentation` is the branch which I made the changes locally
git rebase main

(Regenerate codes)
cd CodeGeneration
swift run generate-swiftsyntax ../Sources 

(Soft reset, commit, and push)
cd ..
git reset --soft main
git commit -m "Documentation - Typos & Consistency Fixes"
git push -f

kimdv commented:
@swift-ci please test windows

ahoppen commented:
https://github.com/apple/swift/pull/66950

@swift-ci Please test

ahoppen commented:
CI is currently failing with an unrelated issue (https://github.com/apple/swift-syntax/pull/1855). I’ll trigger CI agains once that issue has been resolved.

joey-gm commented:
@swift-ci Please test

kimdv commented:
https://github.com/apple/swift/pull/66950

@swift-ci Please test

ahoppen commented:
A few tests in this repo failed, AFAICT mostly because of the “preceed” -> “precede” rename. Could you adjust those failing test cases?

joey-gm commented:
> A few tests in this repo failed, AFAICT mostly because of the “preceed” -> “precede” rename. Could you adjust those failing test cases?

Fixed the typos and also changed the "eofToken" terminology to "endOfFileToken" in Tests/SwiftSyntaxTest/DebugDescriptionTests.swift

Passed all the tests in my local branch (macOS)

kimdv commented:
> > A few tests in this repo failed, AFAICT mostly because of the “preceed” -> “precede” rename. Could you adjust those failing test cases?
> 
> Fixed the typos and also changed the "eofToken" terminology to "endOfFileToken" in Tests/SwiftSyntaxTest/DebugDescriptionTests.swift
> 
> Passed all the tests in my local branch (macOS)

Have a PR open for fixing the `endOfFileToken` in #1860

kimdv commented:
@joey-gm if you rebase this branch it should be fixed

kimdv commented:
@swift-ci please test

ahoppen reviewed (approved):
Thank you 🙏🏽

ahoppen commented:
@swift-ci Please test

ahoppen commented:
https://github.com/apple/swift/pull/66950

@swift-ci Please test macOS

ahoppen commented:
I’m sorry @joey-gm, this PR conflicted with https://github.com/apple/swift-syntax/pull/1878. Could you rebase once more.

joey-gm commented:
> I’m sorry @joey-gm, this PR conflicted with #1878. Could you rebase once more.

@ahoppen Updated and rebased. Please kindly check and test if this resolves the conflict.

ahoppen commented:
@swift-ci Please test

ahoppen commented on Sources/SwiftSyntax/generated/syntaxNodes/SyntaxDeclNodes.swift:
```diff
@@ -3778,7 +3778,7 @@ public struct InitializerDeclSyntax: DeclSyntaxProtocol, SyntaxHashable {
   }
   
-  /// The arguments of the initializer. While the function signature allows specifying an return clause, doing so is not semantically valid.
+  /// The arguments of the initializer. While the function signature allows specifying a return clause, doing so is not semantically valid.
```
This file is being generated. You need to also fix the typo here:

https://github.com/apple/swift-syntax/blob/1d36fc2b6a923a1ae79eede12bdd977632e33815/CodeGeneration/Sources/SyntaxSupport/DeclNodes.swift#L1344

ahoppen reviewed (commented):
CI found one more inconsistency where you only updated a generated file. I hope that we can get this merged soon.

joey-gm commented on CodeGeneration/Sources/SyntaxSupport/DeclNodes.swift:
```diff
@@ -1353,7 +1353,7 @@ public let DECL_NODES: [Node] = [
         name: "Body",
         kind: .node(kind: .codeBlock),
-        documentation: "The initializer’s body. Missing if the initialier is a requirement of a protocol declaration.",
+        documentation: "The initializer’s body. Missing if the initializer is a requirement of a protocol declaration.",
```
Fixed one more typo and regenerated the codes

joey-gm commented:
> CI found one more inconsistency where you only updated a generated file. I hope that we can get this merged soon.

Updated. Please kindly test. :)

ahoppen commented:
@swift-ci Please test

ahoppen commented:
https://github.com/apple/swift/pull/66950

@swift-ci Please test
